packages/file-store/index.ts

```diff
@@ -188,25 +182,29 @@ export class FileStore extends DataStore {
   }
 
   async declareUploadLength(id: string, upload_length: number) {
-    const file = this.configstore.get(id)
+    const file = await this.configstore.get(id)
 
     if (!file) {
       throw ERRORS.FILE_NOT_FOUND
     }
 
     file.size = upload_length
 
-    this.configstore.set(id, file)
+    await this.configstore.set(id, file)
   }
 
   async deleteExpired(): Promise<number> {
     const now = new Date()
     const toDelete: Promise<void>[] = []
 
-    const uploadInfos = this.configstore.all
-    for (const file_id of Object.keys(uploadInfos)) {
+    if (!this.configstore.list) {
+      return 0
```

Comment:
Can we have a more explicit error here (or somewhere else) that expiring uploads is not supported if your config store does not support the list operation?

Or should we make the list operation just mandatory?